File: back/DevC/src/main/java/com/back/devc/domain/interaction/notification/dto/NotificationListResponse.java
package com.back.devc.domain.interaction.notification.dto;

import java.util.List;

public record NotificationListResponse(
        List<NotificationResponse> notifications,
        int page,
        int size,
        long totalElements,
        int totalPages,
        boolean hasNext
) {
    public NotificationListResponse(List<NotificationResponse> notifications) {
        this(notifications, 0, notifications.size(), notifications.size(), 1, false);
    }
}

File: back/DevC/src/main/kotlin/com/back/devc/domain/interaction/notification/dto/NotificationListResponse.kt
package com.back.devc.domain.interaction.notification.dto

data class NotificationListResponse(
    val notifications: List<NotificationResponse>,
    val page: Int,
    val size: Int,
    val totalElements: Long,
    val totalPages: Int,
    val hasNext: Boolean,
) {
    constructor(notifications: List<NotificationResponse>) : this(
        notifications = notifications,
        page = 0,
        size = notifications.size,
        totalElements = notifications.size.toLong(),
        totalPages = 1,
        hasNext = false,
    )
}